src/common/droid-util.c

```diff
@@ -1565,14 +1565,21 @@ void pa_droid_add_card_ports(pa_card_profile *cp, pa_hashmap *ports, pa_droid_ma
     add_ports(core, cp, ports, am, NULL);
 }
 
-static void log_active_quirks(pa_droid_quirks *quirks) {
+void pa_droid_quirk_log(pa_droid_hw_module *hw) {
     uint32_t i;
 
-    if (quirks) {
-        pa_log_debug("Enabled quirks:");
-        for (i = 0; i < sizeof(valid_quirks) / sizeof(struct droid_quirk); i++)
-            if (quirks->enabled[i])
-                pa_log_debug("  %s", valid_quirks[i].name);
+    pa_assert(hw);
+
+    if (hw->quirks) {
+        for (i = 0; i < sizeof(valid_quirks) / sizeof(struct droid_quirk); i++) {
```

Comment:
again matter of taste, but using likes of "sizeof valid_quirks / sizeof *valid_quirks"-> no need to think about type correctness